leetcode/iter_2.rs

```diff
@@ -0,0 +1,39 @@
+//Count Matches
+impl Solution {
+    pub fn number_of_matches(mut n: i32) -> i32 {
+        let mut matches = vec![];
+
+        while n > 1 {
+            matches.push(n / 2);
+
+            if n % 2 == 0 {
+                n /= 2;
+            } else {
+                n /= 2;
+                n += 1;
+            }
+        }
+
+        return matches.iter().fold(0, |sum, x| sum + x);
+    }
+}
+
+//Minimum Number of Moves
+impl Solution {
+    pub fn min_moves_to_seat(mut seats: Vec<i32>, mut students: Vec<i32>) -> i32 {
+        let mut moves = 0;
+
+        seats.sort();
```

Comment:
In soring a vector once, [`sort_unstable`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_unstable) is faster than `sort`.

NOTE: If you don't know about sorting stability, please read [this article](https://www.baeldung.com/cs/stable-sorting-algorithms) (and feel free to ask if you have any question).